packages/npm/financekit/src/index.tsx

```diff
@@ -138,104 +159,531 @@ const CurrentBalance = S.Union(
 )
 
 const AccountBalance = S.Struct({
-  id: S.UUID,
-  accountID: S.UUID,
+  id: uuid,
+  accountID: uuid,
   currentBalance: CurrentBalance,
 })
 
-export async function requestAuthorization(): Promise<
-  typeof AuthorizationStatus.Type
+export const FinancekitError = S.Struct({
+  userInfo: S.OptionFromNullOr(S.Any),
+  nativeStackAndroid: S.OptionFromUndefinedOr(
+    S.Array(
+      S.Struct({
+        lineNumber: int,
+        file: S.String,
+        methodName: S.String,
+        class: S.String,
+      })
+    )
+  ),
+  nativeStackIOS: S.OptionFromUndefinedOr(S.Array(S.String)),
+  domain: S.OptionFromUndefinedOr(S.String),
+  message: S.String,
+  code: S.Literal(
+    '@candlefinance.financekit.account_balances_invalid',
+    '@candlefinance.financekit.account_balance_history_invalid',
+    '@candlefinance.financekit.accounts_invalid',
+    '@candlefinance.financekit.os_version_too_low',
+    '@candlefinance.financekit.params_invalid',
+    '@candlefinance.financekit.query_invalid',
+    '@candlefinance.financekit.transaction_history_invalid',
+    '@candlefinance.financekit.transactions_invalid',
+    '@candlefinance.financekit.unknown',
+    '@candlefinance.financekit.unknown_account_type',
+    '@candlefinance.financekit.unknown_authorization_status',
+    '@candlefinance.financekit.unknown_current_balance_type',
+    '@candlefinance.financekit.unknown_error_response_schema',
+    '@candlefinance.financekit.unknown_response_schema',
+    '@candlefinance.financekit.android_not_supported'
+  ),
+})
+
+const makeError = <Code extends String & (typeof FinancekitError.Type)['code']>(
+  error: any,
+  code: Code
+): Discriminated<typeof FinancekitError.Type, 'code', Code> =>
+  FinancekitError.make({
+    code: code,
+    userInfo: O.none(),
+    message: JSON.stringify(error),
```

Comment:
```suggestion
    message: error.message,
```
I think if for some reason our decoding ever fails here, better to just copy the human-readable message of the underlying error rather then make it an unreadable mess haha